app/models/cinstance.rb

```diff
@@ -199,9 +201,17 @@ def self.by_service(service)
 
   # maybe move both limit methods to their models?
 
-  def self.serialization_preloading
-    includes(:application_keys, :plan, :user_account,
-             service: [:account, :default_application_plan])
+  def self.serialization_preloading(format = nil)
+    # With Rails 6.1 trying to include plan->issuer without service results in
+    #   > Cannot eagerly load the polymorphic association :issuer
+    # When both have the same sub-includes, cache takes care of the duplicate queries.
+    service_includes = %i[proxy account]
+    plan_includes = [{issuer: service_includes}]
+    if format == "xml"
+      service_includes << :default_application_plan
+      plan_includes << :original
+    end
```

Comment:
Is this only for XML responses?... JSON/UI don't need these fields?

Reply:
For JSON it complained that we are eager loading stuff that is not later accessed. UI uses another controller.

Reply:
I don't like this too much... but OK, if that makes the queries more efficient - let's keep it.

Reply:
Otherwise these are queried once per entry displayed in the XML. So we should have it. But if you have a better idea how to do, that would be cool.